### ISSUE TITLE
Feature proposal: Reserved seats

### ISSUE THREAD

**Issue by DrAnax** [Zamiell/hanabi-live#2311]
After #1253 is implemented, I think a nice feature would be to be able to create a table with fixed number of players and reserved seats.

I see that many times people want to play a game with their buddies and use password protected tables just for that.

So, my idea is the following:

1. Be able to create table with a set number of players
2. Be able to reserve seats for specific users
3. Be able to reserve a seat for any player with minimum (or even maximum?) number of games X

When a player joints the table and doesn't meet the requirements he will be forced to become a spectator.

**Comment by Zamiell** [comment]
1 is already done now with the last PR right?

**Comment by DrAnax** [comment]
> 1 is already done now with the last PR right?

yes

**Comment by Zamiell** [comment]
i dont really like 3.
minimum games is a proxy for skill level, but it isn't a very good proxy, which makes the feature questionable.

as for 2, the password feature already accomplishes this goal in a pretty good way, and the feature is easy to understand, both for players, and for people reading the code.

i guess the advantage of 2 is specifically that players do not have to pre-share a password?
which i guess makes playing with your friends less clunky.
but its unclear to me whether or not that benefit is worth it when the password feature already works so well

**Comment by DrAnax** [comment]
> as for 2, the password feature already accomplishes this goal in a pretty good way, and the feature is easy to understand, both for players, and for people reading the code.
> 
> i guess the advantage of 2 is specifically that players do not have to pre-share a password? which i guess makes playing with your friends less clunky. but its unclear to me whether or not that benefit is worth it when the password feature already works so well

Maybe you want to start a public table, but reserve some of the seats for your friends.

**Comment by Zamiell** [comment]
in the hyphen-ated group,
1) everyone usually coordinates in a voice chat before a table is created, so there is very little lag time in which a table would be filled by people who are not already in the voice chat
2) everyone usually does not play "public" games with random strangers joining

a feature for a hypothetical user who is
1) not part of the hyphen-ated group
2) has friends that they always want to play with
3) but also wants to play with public strangers
seems esoteric
